### PR TITLE
Emrge-2163 Adding overwrite functionality

### PR DESCRIPTION
**Pull request by timmockovative**
## What

Adds overwrite functionality to the airbyte connector

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py [hunk 1]
```diff
@@ -115,17 +122,6 @@ def write_parquet(self, messages, stream_name, stream_schema):
 
         buf = pa.BufferOutputStream()
 
-        #build path
-        path_name = ""
```
Comment (JohnMikkelsonOvative) [inline airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py:119]: the `path_name` variable still exists, but the 'build path' code has been deleted - how does the new write functionality allow for pathing, or does it remove it?

File: airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py [hunk 2]
```diff
@@ -145,25 +141,14 @@ def write_csv(self, messages, stream_name, flattening):
 
         df.to_csv(buf)  # filling that buffer
 
-        #build path
-        path_name = ""
```
Comment (JohnMikkelsonOvative) [inline airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py:149]: same as above on `path_name`

File: airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py [hunk 3]
```diff
@@ -261,16 +246,6 @@ def reorder_list_of_dicts_by_schema(records, schema_order):
 
         schema = avro.schema.parse(schema_json)
 
-        path_name = ""
```
Comment (JohnMikkelsonOvative) [inline airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py:264]: same comment on `path_name` functionality

File: airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py (hunk 1 above)
Comment (timmockovative) [reply airbyte-integrations/connectors/destination-azure-blob-storage-python/destination_azure_blob_storage_python/azure.py:119]: Oh, moved the path building over to the stream writer file. Reduced a bit of redundant code and allowed me to use it for the file overwrite code as well. Path name now comes from the parameters

**Comment by timmockovative** [comment]
Moved pathing over from the individual write functions to the streamwriter. The path writing still follows the same structure as before. Added in overwrite functionality. If the sync mode is overwrite, then it will blast the folder location (if it has anything) on the first flush.

>
> Important: This command can only be run by one of the repository admins:
> Usage: ./tools/bin/get_repo_admins.sh <personal_access_token> <repository>

**Pull request merged**